### PR TITLE
Added API Queries and dynamic styling

### PR DESCRIPTION
**Pull request by alonw0**
Fun coding!

One question, by just putting "offset" in the axios params like you did it translated to { offset: "state value" }, is that how it works everywhere in react with states or is it a axios/React Query trick?

### CODE REVIEW

**Comment by romansndlr** [comment]
> Fun coding!
> 
> One question, by just putting "offset" in the axios params like you did it translated to { offset: "state value" }, is that how it works everywhere in react with states or is it a axios/React Query trick?

@alonw0 Hey man! Anything you put under params as a property on an object gets transformed into a query string. So for example, doing { offset: 10 } gets translated to ?offset=10.

File: src/pages/Home.jsx [hunk 1]
```diff
@@ -19,73 +59,79 @@ function Home() {
...
-                    className="nav-link" // Add active if not feed and not tag
+                    onClick={()=> setActiveTag(null)}
+                    className={classNames("nav-link", {active: !activeTag})}
                   >
                     Global Feed
                   </button>
                 </li>
                 <li className="nav-item">
-                  <a className="nav-link active"># {faker.lorem.word()}</a>
+                 {activeTag && <a className="nav-link active"># {activeTag}</a>}
```
Comment (romansndlr) [inline src/pages/Home.jsx]: You probably want to get rid of the entire `li` if there is no `activeTag`.

File: src/pages/Home.jsx [hunk 2]
```diff
@@ -19,73 +59,79 @@ function Home() {
...
                 </li>
               </ul>
             </div>
+            {articlesQuery.isFetching && <div className="article-preview">Loading articles..</div>}
```
Comment (romansndlr) [inline src/pages/Home.jsx]: Please move below all of the articles, we don't want the entire list to jump when the loading indicator pops out.

File: src/pages/Home.jsx [hunk 3]
```diff
@@ -19,73 +59,79 @@ function Home() {
...
+            {(articles?.length === 0 && articlesQuery.isFetched) && <div className="article-preview">No articles to show... Wanna be the first to post?</div>}
```
Comment (romansndlr) [inline src/pages/Home.jsx]: Nice touch! A few small things:

- Let's move this to the top before we try and render any articles. 
- Let's use `isEmpty` from 'lodash-es` to check that the article list is empty.

File: src/pages/Home.jsx [hunk 4]
```diff
@@ -19,73 +59,79 @@ function Home() {
...
+              <div className="tag-list">
+      
+            {tags?.map((tag) => { return (
+              <a onClick={()=> setActiveTag(tag)} href="#" className="tag-pill tag-default">
+              {tag}
             </a>
+            )})}
+            </div>
+            {tagsQuery.isFetching && <div>Loading Tags...</div>}
+            {(tags?.length === 0 && tagsQuery.isFetched) && <div>No tags are here... yet.</div>}
```
Comment (romansndlr) [inline src/pages/Home.jsx]: Please use `isEmpty` from `lodash-es`.

**Review by romansndlr — CHANGES_REQUESTED** [review]
Amazing work dude! 👏🏻

There are a lot of formatting issues so please make sure you have https://prettier.io/ integrated into your editor and that you run it on every save.

File: src/pages/Home.jsx (hunk 1 above)
Comment (alonw0) [reply src/pages/Home.jsx]: Fixed, please check

File: src/pages/Home.jsx (hunk 2 above)
Comment (alonw0) [reply src/pages/Home.jsx]: Fixed, please check

File: src/pages/Home.jsx (hunk 3 above)
Comment (alonw0) [reply src/pages/Home.jsx]: Fixed, please check

File: src/pages/Home.jsx (hunk 4 above)
Comment (alonw0) [reply src/pages/Home.jsx]: Fixed, please check.

1. For some reason prettier wont wort on auto save with VSCode just in this spesific project
2. I ran an 'npx prettier --write *' on the project folder for temp fix
3. For my first question - i understand the axios params as key/value but we only give it "offset" and not a real key/value base object. seems like it knows to transform it to one, is it for any variable or just for react states?

Comment (romansndlr) [reply src/pages/Home.jsx]: Yeah, we went over this at the meeting tomorrow, there is a shorthand that was introduced with es6 where object properties that have a value with the same name can be collapsed.

**Pull request closed without merging**